### PR TITLE
[RPC]Adding `getspendablebalance` RPC Call.

### LINKED ISSUE #496: Create new rpc call to get total balance of all wallet amounts
Linked from #474

### PR DESCRIPTION
**Pull request by MatWaller**
Adding `getspendablebalance` RPC Call.
- Returns the combined total of all spendable coins.

Closes #496 

autofill
Helpmsg
Working

Bounty Address
```
sv1qqpdsfcqvfcljtcq8txc8kc2dw4e9upj69wnnfv7j0ufcsrv7trs9lqpqfscke6jj92aj4aup387ggk7mathznr8hz3zde6ut87jdwphrns7qqqq00jn8w
```

### CODE REVIEW

File: src/wallet/rpcwallet.cpp [hunk 1]
```diff
@@ -1101,6 +1101,38 @@ static UniValue getbalance(const JSONRPCRequest& request)
...
+    CWallet *const pwallet = wallet.get();
+
+    if (!EnsureWalletIsAvailable(pwallet, request.fHelp)) {
+        return NullUniValue;
+    }
+
+    if (request.fHelp || (request.params.size() > 0 )){
+        throw std::runtime_error(
+            std::string(
+                "getSpendableBalance ()\n"
```
Comment (blondfrogs) [inline src/wallet/rpcwallet.cpp]: this name should be the same as the rpc call. No need to make it be camelcase

Comment (MatWaller) [reply src/wallet/rpcwallet.cpp]: Will get this changed now.

Thanks.

Comment (MatWaller) [reply src/wallet/rpcwallet.cpp]: Updated and squashed.

**Comment by xo-orca** [comment]
QA passed on W10, Linx and Mac. Looks good mate!

xo.

**Pull request merged**